### DISCUSSION TITLE
Invoke the before_update on specific column

### DISCUSSION THREAD

**Discussion by Gautam3AI** [sqlalchemy/sqlalchemy#11277] [Usage Questions] [answered]
I have a basic schema construct
```python3
class Table1(SQLDeclarativeBase):
    __tablename__ = "table1"

    id = Column(
        UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()")
    )
    t1_col1 = Column(String, nullable=False)
    t1_col2 = Column(String, nullable=False)

class Table2(SQLDeclarativeBase):
    __tablename__ = "table2"

    id = Column(
        UUID(as_uuid=True), primary_key=True, server_default=text("gen_random_uuid()")
    )
    t2_col1 = Column(Boolean, default=False)
    t2_fk = Column(
        UUID(as_uuid=True), ForeignKey("table1.id", ondelete="CASCADE"), nullable=True
    )

engine = create_engine(db_uri)
Base.metadata.create_all(engine)
Session = sessionmaker(bind=engine)
```

I have an event listener that before the update on Table1's row, it should invoke `update_t2_col1()`
```python3
@event.listens_for(Table1, 'before_update')
def update_t2_col1(mapper, connection, target):
    session = Session()
    table2_row = session.query(Table2).filter(Table2.id == target.id).first()
    table2_row.t2_col1 = True
    session.commit()
```

But I want to invoke this `update_t2_col1()` only on certain column updates, let's say only when `t1_col2` row of `Table1` is being updated. 
Any help how to do this.

Please ignore the sytactical error

**Comment by zzzeek** [comment, marked as answer]
use a conditional against the column itself:

```py
@event.listens_for(Table1, 'before_update')
def update_t2_col1(mapper, connection, target):
    if inspect(target).attrs.t2_col1.history:
          # code
```

however I would **strongly** advise against creating a new Session, transaction, and commiting that whole transaction, in this event.  That's completely wrong.   Simply emit your UPDATE statement on the connection given, stay within that single transaction unless you have something very exotic and specific going on:

```
@event.listens_for(Table1, 'before_update')
def update_t2_col1(mapper, connection, target):
    if inspect(target).attrs.t1_col1.history:
        connection.execute(
            update(Table2).where(Table2.id == target.id).values(t2_col1=True)
        )

```

**Reply by Gautam3AI** [reply]
Thanks for the assistance and demonstrating the correct approach of doing this.
For those who may seek this information in the future, `inspect(target).attrs.t2_col1.history` yields a named tuple [History](https://github.com/sqlalchemy/sqlalchemy/blob/483edf651f6c3db6729f8d75265241b67ccd6b80/lib/sqlalchemy/orm/attributes.py#L2310), with numerous additional functionalities at hand.
